debug/fetch_data.c

```diff
@@ -27,57 +27,164 @@ static const char* getfield(char* line, int num)
 
 #define NCHANNELS 32
 #define NSESSIONS 8
+#define EVLEN 150
+
+typedef sax_word sax_signal[NCHANNELS];
+
+struct sax_signal_set {
+    sax_signal *series;
+    size_t n_series;
+};
+
+struct session {
+    double **series;
+    size_t series_count;
+};
+
+#define PUSH_BACK(series, size, value) \
+    (series) = realloc((series), (++(size)) * sizeof(*series)); \
```

Comment:
- don't use a macro
- check the return value of realloc
- only assign on success